LoongArch-ELF-ABI-EN.adoc

```diff
@@ -0,0 +1,206 @@
+= LoongArch ELF ABI specification
+Loongson Technology Corporation Limited
+V1.00
+
+:encoding: UTF-8
+:frame: top
+:grid: rows
+:valign: middle
+
+== Register Convention
+.Integer Register Convention
+[width="80%",options="header",cols="2m,2m,^5,^3"]
+|====================================================================
+|Name     |Alias        |Meaning             |Preserved across calls
+|$r0      |$zero        |Constant zero       |--
+|$r1      |$ra          |Return address      |No
+|$r2      |$tp          |Thread pointer      |--(Unallocatable)
+|$r3      |$sp          |Stack pointer       |Yes
+|$r4-$r11 |$a0-$a7      |Argument registers  |No
+|$r4-$r5  |$v0-$v1      |Return value        |No
+|$r12-$r20|$t0-$t8      |Temp registers      |No
+|$r21     |$x           |Reserved            |--(Unallocatable)
+|$r22     |$fp          |Frame pointer       |Yes
```

Comment:
Is maintaining correct frame pointer mandatory in the LA ABI? Maybe we need some clarification here (or below the table).

Reply:
Sure it is, $r22 is a Preserved register actually, meaning if ask gcc for generating code with FRAME POINTER(-fno-omit-frame-pointer), gcc will use $r22 to save the pointer of stack frame.

Reply:
> Sure it is, $r22 is a Preserved register actually, meaning if ask gcc for generating code with FRAME POINTER(-fno-omit-frame-pointer), gcc will use $r22 to save the pointer of stack frame.

I mean whether it's conformant for a program to trash this register for an extra saved register, i.e. the `-fomit-frame-pointer` behavior? If so, there might be implications (`$fp` is not guaranteed to contain a valid frame pointer at all times; we might have to give `$fp` an alias such as `$s9`, etc.), and we need to clarify this.

Reply:
thanks, we will add the clarification.